app/lib/lti_advantage/lti_user.rb

```diff
@@ -44,6 +46,17 @@ def email
       end
     end
 
+    # If a LTI 1.1 install is being migrated to an Advantage install,
+    # pre-existing users won't be found correctly. So we swap out their
+    # lti id with their new lti_user_id before proceeding
+    def _migrate_lti_user
+      user = User.find_by(lti_user_id: @lti_token[LtiAdvantage::Definitions::LTI11_LEGACY_USER_ID_CLAIM])
+      if user
+        user.lti_user_id = @lti_user_id
```

Comment:
Being a destructive operation here, would it make sense to backup this data into a new field so we don't permanently loose the data? Something like `legacy_lti_user_id`? Or `old_lti_user_id`? And after it is backed up, then set the new id on `lti_user_id`?

Just a thought I had.

Reply:
The `legacy_lti_user_id` actually comes over in every lti launch and is saved to the model [here](https://github.com/atomicjolt/lti_starter_app/blob/ae8d035dcd8b330dbf500ba090bb371cbf8dade2/app/lib/lti_advantage/lti_user.rb#L21).
We can make sure to add it here as well, but it won't make a difference functionally

Reply:
oh that's right. I forgot we had that field. lol. k, carry on. :]